data_analysis/model_ad/notebooks/MG-40_Create_harmonized_pathology_source_file.rmd

```diff
@@ -0,0 +1,713 @@
+
+
+## Generate model-specific intermediate files
+File-measurement-unit and display value mappings for legacy models are from the v1 data spreadsheet here:
+https://docs.google.com/spreadsheets/d/1B3ZMReC7nR2CGgpFNJuxJTj__lF2h6dt/edit?gid=1108411940#gid=1108411940
+
+### 3xTG-AD
+```{r}
+threex_pathology <- threex_pathology_source %>%
+
+  # Filter out rows with unused 'Stain' values
+  filter(
+    stain %in% stain_keeps
+
+    # Pivot targeted measurement columns; capture colname as 'unit' and value as 'measurement'
+  ) %>% pivot_longer(
+  cols = c(objectDensity, averageObjectVolume, totalObjectVolume),
+  names_to = "units",
+  values_to = "measurement",
+  values_transform = list(measurement = as.numeric)
+
+  # Convert units to display values
+) %>% mutate(
+  units = case_match(
+    units,
+    'objectDensity' ~ objectDensity_display_name,
+    'averageObjectVolume' ~ averageObjectVolume_display_name,
+    'totalObjectVolume' ~ totalObjectVolume_display_name
+  )
+
+  # Map 'stain' values to 'type' display names
+) %>% mutate(
+  type = case_when(
+    stain == 'IBA1' ~ IBA1_display_name,
+    stain == 'GFAP' ~ GFAP_display_name,
+    stain == 'S100B' ~ S100B_display_name,
+    stain == 'LAMP1' ~ LAMP1_display_name,
+    stain == 'AT8' ~ AT8_display_name,
+    stain == 'HT7' ~ HT7_display_name,
+    stain == 'ThioS' & units == averageObjectVolume_display_name ~ ThioS_size_display_name,
+    stain == 'ThioS' & units == objectDensity_display_name ~ ThioS_density_display_name,
+    # Drops ThioS totalObjectVolume measurements
+    .default = 'UNMAPPED_DROPME'
+  )
+
+  # drop rows for unused ThioS stain/unit combinations
+) %>% filter(
+  type != 'UNMAPPED_DROPME'
+
+  # Drop rows with NA measurements
+) %>% filter(
+  !is.na(measurement)
+
+  # Populate model column
+) %>% add_column(
+  model = '3xTG-AD'
+
+  # Drop unwanted columns
+) %>% select(
+  all_of(keeps_with_pool_join_field)
+)
+
+threex_pathology
+```
+
+
+### 5xFAD
+```{r}
+# Filter each source to remove unused rows by 'Stain' value
+fivex_gfap_s100b <- fivex_gfap_s100b_source[fivex_gfap_s100b_source$Stain %in% c('GFAP', 'S100B'),]
+fivex_lamp1 <- fivex_lamp1_source[fivex_lamp1_source$Stain == 'LAMP1',]
+fivex_iba1_thios <- fivex_iba1_thios_source[fivex_iba1_thios_source$Stain %in% c('Iba1', 'ThioS'),]
+
+# Pivot targeted measurement columns; capture colname as 'unit' and value as numeric 'measurement'
+# GFAP/S100B source file
+fivex_gfap_s100b_pivot <- fivex_gfap_s100b %>%
+  pivot_longer(
+    cols = c('X.objects..sqmm', 'mean.object.area..sqmm.'),
+    names_to = "units",
+    values_to = "measurement",
+    values_transform = list(measurement = as.numeric)
+
+    # Rename columns for consistency
+  ) %>% rename(c('type' = 'Stain',
+                 'individualID' = 'IndividualID',
+                 'specimenID' = 'SpecimenID')
+
+               # Populate model column
+) %>% add_column(
+  model = '5xFAD'
+
+  # Drop unused columns
+) %>% select(keeps_with_join_fields)
```

Comment:
I'm surprised this works since `keeps_with_join_fields` is a list of characters. I thought you always needed `all_of` for that. Does R not get mad about it when you run this?

Reply:
I added the missing all_of in three spots where I found it missing (~@ lines 299, 322, & 345)